source/crud/tests/unified/estimatedDocumentCount.yml

```diff
@@ -0,0 +1,149 @@
+description: "estimatedDocumentCount"
```

Comment:
I think we should add a test that asserts that any server error that is NOT NamespaceNotFound is correctly reported (I had a bug in my implementation of this that wouldn't have been caught by any of these tests).  We can do this with a failpoint fairly easily I think.

Reply:
Sounds good! Added two new tests, one for 4.9+ and < 4.9.

Reply:
Can you add one more test: one where the failpoint doesn't force the connection to close, but rather replies with an error code that is not NamespaceNotFound?  

Rationale is that some drivers will be catching CommandException explicitly, and then ignoring NamespaceNotFound.  With the error that just closes the connection, that code is skipped entirely since a SocketException is what will be thrown instead.

Reply:
Good point! Added. Now there are four tests for correct error handling: socket error in 4.9+ and < 4.9, and command error in 4.9+ and < 4.9.

Reply:
Thanks for adding these.  All passing in Java driver.